### DISCUSSION TITLE
如何禁止v2ray进行远程域名解析

### DISCUSSION THREAD

**Discussion by aries1998** [v2fly/v2ray-core#2471] [Q&A]
比如某个域名domain1, 本地解析地址为ip1,   在服务端机子上解析地址为ip2 
那么访问domain1的时候预期是访问的ip1 ， 但是由于v2ray貌似是会再远程解析一次dns，变成实际访问的是ip2， 用ss没这个问题。

本地客户端用的v2rayA， 设置里面防止dns污染设置是关。

**Comment by mydogshitgold** [comment]
It appears that the v2ray on your end  or the remote server is engaged in domain name sniffing, a process whereby it uses information in the packet, such as the SNI or HTTP Host header, to replace the host field of outbound requests with domain names, rather than IP addresses. 

To prevent this behavior, you can simply disable the sniffing feature by setting `sniffing.enabled` to `false`.

Note if disabling sniffing on your end does not work. Then the remote server may have the sniffing feature enabled. To test whether this is the case, you can try to fetch a website with an invalid IP address, for example, by using the following command:

```bash
curl https://google.com --resolve google.com:443:1.1.1.1
```

if it outputs:
```html
<HTML><HEAD><meta http-equiv="content-type" content="text/html;charset=utf-8">
<TITLE>301 Moved</TITLE></HEAD><BODY>
<H1>301 Moved</H1>
The document has moved
<A HREF="https://www.google.com/">here</A>.
</BODY></HTML>
```

instead of:
```
curl: (60) SSL: no alternative certificate subject name matches target host name 'google.com'
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
```

Then it's certain that the server also enabled sniffing, which is no way to disable if you don't
have control over the server.

**Comment by aries1998** [comment]
Thanks  it works.
The v2ray server installed with x-ui set the sniffing.enabled=true by default .
